### ISSUE TITLE
NoMethodError in admin/variant_overrides#index

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#4249]
## Error in OpenFoodNetwork

**NoMethodError** in **admin/variant_overrides#index**
undefined method `has_spree_role?' for nil:NilClass

[View on Bugsnag](https://app.bugsnag.com/yaycode/openfoodnetwork/errors/5d75e17d4772cd00197f6f77?event_id=5d75e17d004f2eafe17c0000&i=gh&m=ci)

## Stacktrace

    app/models/enterprise.rb:180 - block in <class:Enterprise>
    lib/open_food_network/permissions.rb:192 - managed_enterprises
    lib/open_food_network/permissions.rb:29 - variant_override_hubs
    app/controllers/admin/variant_overrides_controller.rb:46 - load_data

[View full stacktrace](https://app.bugsnag.com/yaycode/openfoodnetwork/errors/5d75e17d4772cd00197f6f77?event_id=5d75e17d004f2eafe17c0000&i=gh&m=ci)

*Created by Maikel via Bugsnag*

**Comment by agustinariq** [comment]
Hello! Can I work on this issue?

**Comment by RachL** [comment]
@agustinariq of course, you are very welcome to! However it might be something a bit difficult to tackle for a first issue. Don't hesitate to join us [on Slack](https://join.slack.com/t/openfoodnetwork/shared_invite/enQtNzY3NDEwNzM2MDM0LWFmNGRhNDUwYzNmNWNkYmFkMzgxNDg1OTg1ODNjNWY4Y2FhNDIwNmE4ZWI0OThiMGNmZjFkODczNGZiYTJmNWI) if you have any question!

**Comment by agustinariq** [comment]
@RachL Great! What I've done so far is to create a method in the ApplicationController to redirect unauthenticated users to the login page and then call it in a before_filter in the problematic controller.

What do you think?

**Comment by RachL** [comment]
@agustinariq Thanks! I'm afraid I would be able to help you as I'm not a dev :) But maybe one of our @openfoodfoundation/developers can help you.

**Comment by luisramos0** [comment]
hi @augnustin
I assume this is your attempt: https://github.com/agustinariq/openfoodnetwork/commit/003dc47792dd95c6d7d285a9217fd7975364312b

I think this will solve the problem, but I think we need to understand the problem.

I think this problem is only occurring in non spree controllers that inherit from  /admin/resource_controller
Pages under spree/admin/ that inherit from spree/admin/resource_controller (overridden in OFN) do not have this problem, for example the URL admin/products works perfectly.

Anyway, I think the solution should be added to 
Spree::Admin::BaseController.class_eval

**Comment by mkllnk** [comment]
Hi @augnustin!

Another really good strategy to make sure that it works is to first write a test that fails in the same way as described here in the issue. Then you apply your fix and see if it works. In this case, a controller spec would be good.

**Comment by mkllnk** [comment]
This issue has been linked to an error in Bugsnag
[**NoMethodError** in **admin/variant_overrides#index**](https://app.bugsnag.com/yaycode/openfoodnetwork-aus/errors/62a172d6a7e2010008753a87?event_id=62a6c05c009514a9aad40000&i=gh&m=li)

**Comment by RachL** [comment]
closing this one as we won't fix inventory issues anymore
